eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -105,7 +106,7 @@ class PeerConnection(nodeParams: NodeParams, switchboard: ActorRef, router: Acto
           // off basic_mpp and payment_secret. As long as they're provided in the invoice it's not an issue.
           // We use a long enough mask to account for future features.
           // TODO: remove that once eclair-mobile is patched.
-          val tweakedFeatures = BitVector.bits(nodeParams.features.bits.reverse.toIndexedSeq.zipWithIndex.map {
+          val tweakedFeatures = BitVector.bits(nodeParams.features.toByteVector.bits.reverse.toIndexedSeq.zipWithIndex.map {
```

Comment:
You can now make this code a lot more readable. Instead of modifying a `ByteVector`, just modify the `Features` object to remove the `basic_mpp` and `payment_secret` features from the set of activated features.

Reply:
Yes, this is done in 962d2a5 and it's been extracted into a function.